api/v1alpha1/machine.go

```diff
@@ -49,6 +49,25 @@ const (
 type MachineSpec struct {
 	// Connection contains connection data for a Baseboard Management Controller.
 	Connection Connection `json:"connection"`
+
+	// ProviderOptions contains provider specific options.
+	// +optional
+	ProviderOpts ProviderOpts `json:"providerOpts,omitempty"`
+}
+
+type ProviderOpts struct {
+	// RPC contains rpc provider options.
+	// +optional
+	RPC OptRPC `json:"rpc"`
```

Comment:
What's motivated the `Opt*` format instead of `RPCOpts`? Its a little peculiar to have `*Opts` and `Opt*`.

Reply:
The idea was taken from how i've custom errors defined. The struct ends in `Error` and the custom error starts with `Err`. stdlib [example](https://github.com/golang/go/blob/399b2a4b1b7857444c38305025cc793c9377e415/src/strconv/atoi.go#L21-L28). I don't know if I have a strong preference here. Have an alternative/preference, by chance?

```Go
// ErrSyntax indicates that a value does not have the right syntax for the target type.
var ErrSyntax = errors.New("invalid syntax")

// A NumError records a failed conversion.
type NumError struct {
	Func string // the failing function (ParseBool, ParseInt, ParseUint, ParseFloat, ParseComplex)
	Num  string // the input
	Err  error  // the reason the conversion failed (e.g. ErrRange, ErrSyntax, etc.)
}
```

Reply:
Updated the naming. Let me know what you think. Thanks!